src/packages/sdk/src/cli/getSchema.ts

```diff
@@ -11,88 +12,172 @@ const readFile = promisify(fs.readFile)
  */
 
 export async function getSchemaPath(
-  schemaPathFromArgs?,
+  schemaPathFromArgs?: string,
+): Promise<string | null> {
+  return getSchemaPathInternal(schemaPathFromArgs, {
+    cwd: process.cwd(),
+  })
+}
+
+export async function getSchemaPathInternal(
+  schemaPathFromArgs?: string,
+  opts: { cwd: string } = {
+    cwd: process.cwd(),
+  },
 ): Promise<string | null> {
   if (schemaPathFromArgs) {
-    // try the user custom path
+    // 1. try the user custom path
     const customSchemaPath = await getAbsoluteSchemaPath(
       path.resolve(schemaPathFromArgs),
     )
-    if (customSchemaPath) {
-      return customSchemaPath
-    } else {
+    if (!customSchemaPath) {
       throw new Error(
         `Provided --schema at ${schemaPathFromArgs} doesn't exist.`,
       )
     }
+
+    return customSchemaPath
+  }
+
+  // 2. Try the package.json `prisma.schema` custom path
+  // 3. Try the conventional ./schema.prisma or ./prisma/schema.prisma paths
+  // 4. Try resolving yarn workspaces and looking for a schema.prisma file there
+  const schemaPath =
+    (await getSchemaPathFromPackageJson(opts.cwd)) ??
+    (await getRelativeSchemaPath(opts.cwd)) ??
+    (await resolveYarnSchema(opts.cwd))
+
+  if (schemaPath) {
+    return schemaPath
+  }
+
+  return null
+}
+
+export async function getSchemaPathFromPackageJson(
+  cwd: string,
+): Promise<string | null> {
+  const pkgJson = await readPkgUp({ cwd })
+  const schemaPathFromPkgJson = pkgJson?.packageJson?.prisma?.schema as unknown
+
+  if (!schemaPathFromPkgJson || !pkgJson) {
+    return null
   }
 
-  // try the normal cwd
-  const relativeSchemaPath = await getRelativeSchemaPath(process.cwd())
-  if (relativeSchemaPath) {
-    return relativeSchemaPath
+  if (typeof schemaPathFromPkgJson !== 'string') {
+    throw new Error(
+      `Provided schema path configuration \`${schemaPathFromPkgJson}\` at ./${path.relative(
+        cwd,
+        pkgJson.path,
+      )} must be of type string`,
+    )
   }
 
-  // in case no schema can't be found there, try the npm-based INIT_CWD
-  if (process.env.INIT_CWD) {
-    return (
-      (await getRelativeSchemaPath(process.env.INIT_CWD)) ||
-      (await resolveYarnSchema())
```

Comment:
I removed the INIT_CWD logic. As seen with Tim, it looks like INIT_CWD is always process.cwd() anyway.
> If you want your script to use different behavior based on what subdirectory you’re in, you can use the INIT_CWD environment variable, which holds the full path you were in when you ran npm run

See the documentation full doc here: https://docs.npmjs.com/cli/run-script